libs/sdk-common/src/input_parser.rs

```diff
@@ -188,6 +186,15 @@ const BIP353_PREFIX: &str = "bitcoin:";
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+) -> Result<InputType> {
+    let rest_client: Arc<dyn RestClient> = Arc::new(ReqwestRestClient::new()?);
+    parse_with_rest_client(rest_client.as_ref(), input, external_input_parsers).await
+}
+
+pub async fn parse_with_rest_client<C: RestClient + ?Sized>(
```

Comment:
It seems that this function only exists so we can test the input parser by passing a rest client?
Perhaps we can make it with pub(crate) visibility ?

Reply:
It will be used also in the Nodeless SDK, so needs to be pub